repositories/src/main/scala/cromwell/pipeline/datastorage/dto/FileParameter.scala

```diff
@@ -0,0 +1,43 @@
+package cromwell.pipeline.datastorage.dto
+
+import org.mongodb.scala.Document
+import play.api.libs.json._
+
+final case class FileParameter(name: String, typedValue: TypedValue)
+
+object FileParameter {
+  import OptionFormat._
+
+  implicit object InputNodeFormat extends Format[FileParameter] {
```

Comment:
This formatter is quite sophisticated and looks strange.
Why does it contain so much of a logic? Do you want to somehow "remember" the type of the value, to restore it safely from `json`?

Reply:
Formatter for ADT now merged into develop. You can use it instead of implementing all of this on your own.

Reply:
Yep, Thank you.